tests/test_ssl.py

```
@@ -2997,6 +3095,248 @@ def test_wantWriteError(self) -> None:
 
     # XXX want_read
 
+    def _attempt_want_write_error(
+        self, client: Connection, buffer_size: int
+    ) -> bytes:
+        """
+        Attempts to send application data over SSL to trigger WantWriteError.
+        Returns successful_size if triggered,
+        otherwise calls pytest.fail.
+        """
+        logger.debug("--- Phase 1: Attempting to trigger WantWriteError ---")
+        initial_want_write_triggered = False
+        max_num_of_attempts = 100000
+
+        for i in range(max_num_of_attempts):
+            msg = b"Y" * buffer_size
+            try:
+                client.send(msg)
+                logger.debug(
+                    "_attempt_want_write_error() trying "
+                    f"to send {i + 1}th message"
+                )
+            except SSL.WantWriteError:
+                logger.debug(
+                    f"After {i + 1} attempt(s) successfully induced a "
+                    f"WantWriteError with message size {buffer_size}. "
+                    "Buffer location in _attempt_want_write_error() "
+                    f"is {id(msg):#x}"
+                )
+                initial_want_write_triggered = True
+                break  # Exit loop as desired error was triggered
+            except Exception as e:  # pragma: no cover
```

In general, we strongly avoid no cover pragmas. You can simply omit this handling, if some other exception is raised, you can just let it propagate.

---

Good to know. Have revised to remove all but one of my no cover pragmas. The one remaining in _shutdown_connections() is forgivable I think but see what you think.